packages/application-components/src/experimental/hooks/use-debounced-promise-callback/use-debounced-promise-callback.ts

```diff
@@ -1,9 +1,15 @@
 import debounce from 'debounce-promise';
 import { useCallback } from 'react';
 
-const useDebouncedPromiseCallback = <T extends (...args: any[]) => any>(
-  promiseCallback: T,
-  wait: number
-) => useCallback(debounce(promiseCallback, wait), [promiseCallback]);
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type DebouncedCallback<T> = (...args: any[]) => Promise<T>;
```

Comment:
I tried `unknown` but I got a pretty tricky TS error on the consumer side, which I spent too much time trying to resolve.
Saving this for another day

![image](https://user-images.githubusercontent.com/462507/101622953-0310f080-3a18-11eb-9097-5a02af38d4fd.png)